shippers/composite.py

```diff
@@ -30,3 +30,7 @@ def send(self, event: dict[str, Any]) -> Any:
     def flush(self) -> None:
         for shipper in self._shippers:
             shipper.flush()
+
+    def discover_dataset(self, event: dict[str, Any]) -> None:
```

Comment:
`discover_dataset` should be defined only in the concrete `ElasticsearchShipper` and called on it, not on `CompositeShipper`